core/proto/core.proto

```diff
@@ -40,4 +39,30 @@ message Certificate {
   optional int64 expires = 6; // Unix timestamp (nanoseconds)
 }
 
+message Registration {
+        optional int64 id = 1;
+        optional bytes key = 2;
+        repeated string contact = 3;
+        optional bool contactsPresent = 4;
+        optional string agreement = 5;
+        optional bytes initialIP = 6;
+        optional int64 createdAt = 7; // Unix timestamp (nanoseconds)
+        optional string status = 8;
+}
+
+message Authorization {
+        optional string id = 1;
+        optional string identifier = 2;
+        optional int64 registrationID = 3;
+        optional string status = 4;
+        optional int64 expires = 5; // Unix timestamp (nanoseconds)
+        repeated core.Challenge challenges = 6;
+        optional bytes combinations = 7;
+}
+
+message CertificateRequest {
```

Comment:
It looks like you're trying to share this request object between SA and CA, but I don't think that's a good approach. I think for each RPC Foo in each service, we have a `FooRequest` object in that service's proto. That way we don't excessively entangle things.